### PR TITLE
[CFM-543] Auto log-in invited users.

### PR DESCRIPTION
**Pull request by ybaras**
[CFM-543](https://issuetracker.amplexor.com/jira/browse/CFM-543)

### CODE REVIEW

File: project/profiles/capacity4more/modules/c4m/og/c4m_og_invite_users/c4m_og_invite_users.module [hunk 1]
```diff
@@ -29,6 +29,79 @@ function c4m_og_invite_users_menu_alter(&$items) {
...
+
+      return FALSE;
+    }
+
+    return TRUE;
+  }
+
+  // No user logged in. Log in the invited user.
+  $account = array('uid' => $membership->etid);
+  user_login_submit(array(), $account);
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og_invite_users/c4m_og_invite_users.module]: we probably do not want to login inside the access callback

**Comment by ybaras** [comment]
**No user logged in = Auto login.**

no-login

**Comment by ybaras** [comment]
**The invited user is already logged in**

anim

**Comment by ybaras** [comment]
**A different user is logged in**

anim

File: project/profiles/capacity4more/modules/c4m/og/c4m_og_invite_users/c4m_og_invite_users.module (hunk 1 above)
Comment (ybaras) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og_invite_users/c4m_og_invite_users.module]: The issue calls for auto-login functionality. Where should it be implemented?

Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og_invite_users/c4m_og_invite_users.module]: on the page callback ?

File: project/profiles/capacity4more/modules/c4m/og/c4m_og_invite_users/c4m_og_invite_users.module [hunk 2]
```diff
@@ -45,26 +71,31 @@ function c4m_og_invite_users_menu_alter(&$items) {
  *   The membership object of FALSE on failure,
  */
 function c4m_og_invite_users_get_membership_from_hash($hash) {
-  $result = FALSE;
+  $result = &drupal_static(__FUNCTION__ . $hash, array());
+
+  if (isset($result['ogm'])) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og_invite_users/c4m_og_invite_users.module]: rename the `ogm`

**Pull request merged**